File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.ts
function lengthOfLongestSubstring(s: string): number {

    if (s.length === 1){
        return 1
    }

    let result = ""
    let comparation = ""

    for (let letter of s){

        if (comparation.length > result.length){
            result = comparation
        }

        if (!comparation.includes(letter)){
            comparation += letter;
        }
        else{
            comparation = comparation.slice(comparation.indexOf(letter) + 1) + letter;
        }
    }
    
    return Math.max(result.length, comparation.length)
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        longestComparation = ""
        longestResult = ""
        
        for letter in s:
            if len(longestResult) < len(longestComparation):
                    longestResult = longestComparation
            
            if letter not in longestComparation:
                longestComparation += letter
                
            else:
                longestComparation = longestComparation[longestComparation.index(letter) + 1:] + letter
                
                    
        return max(len(longestComparation), len(longestResult))                